pkg/snowflake/db.go

```diff
@@ -282,3 +282,79 @@ func (db *DB) RecreateTableOnMaterializationTypeMismatch(ctx context.Context, as
 
 	return nil
 }
+
+func (db *DB) PushColumnDescriptions(ctx context.Context, asset *pipeline.Asset) error {
+	tableComponents := strings.Split(asset.Name, ".")
+	var schemaName string
+	var tableName string
+	switch len(tableComponents) {
+	case 2:
+		schemaName = strings.ToUpper(tableComponents[0])
+		tableName = strings.ToUpper(tableComponents[1])
+	case 3:
+		schemaName = strings.ToUpper(tableComponents[1])
+		tableName = strings.ToUpper(tableComponents[2])
+	default:
+		return nil
+	}
+	anyColumnHasDescription := false
+	colsByName := make(map[string]*pipeline.Column, len(asset.Columns))
+
+	for _, col := range asset.Columns {
+		colsByName[col.Name] = &col
+		if col.Description != "" {
+			anyColumnHasDescription = true
+		}
+	}
+
+	if asset.Description == "" && (len(asset.Columns) == 0 || !anyColumnHasDescription) {
+		return errors.New("no metadata to push: table and columns have no descriptions")
+	}
+
+	queryStr := fmt.Sprintf(
+		`SELECT COLUMN_NAME, COMMENT 
+          FROM %s.INFORMATION_SCHEMA.COLUMNS 
+          WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME = '%s'`,
+		db.config.Database, schemaName, tableName)
+
+	rows, err := db.Select(ctx, &query.Query{Query: queryStr})
+	if err != nil {
+		return errors.Wrapf(err, "failed to query column metadata for %s.%s", schemaName, tableName)
+	}
+
+	existingComments := make(map[string]string)
+	for _, row := range rows {
+		columnName := row[0].(string)
+		comment := ""
+		if row[1] != nil {
+			comment = row[1].(string)
+		}
+		existingComments[columnName] = comment
+	}
+
+	// Find columns that need updates
+	for _, col := range asset.Columns {
+		if col.Description != "" && existingComments[col.Name] != col.Description {
```

Comment:
this seems like it would add comments only to the columns that already have existing comments, which means it wouldn't add new comments to cols that didn't have them before, no? 
```suggestion
		if existingComments[col.Name] != col.Description {
```

Reply:
col.description is the , description in the asset not the description in the snowflake